src/meltano/core/state_store/azure.py

```diff
@@ -2,9 +2,42 @@
 from __future__ import annotations
 
 import re
+from contextlib import contextmanager
+
+from cached_property import cached_property
```

Comment:
Using `functools.cached_property` when it is available instead of the backport helps mypy determine the type of the property.

```suggestion
import re
import sys
from contextlib import contextmanager

if sys.version_info >= (3, 8):
    from functools import cached_property
else:
    from cached_property import cached_property
```